Add option for disabling nerfing portal-based mob farms

gillesbraun opened the pull request:
Adds a configuration option `game-mechanics.nerf-portal-mob-farms` defaulting to true (current behavior), which prevents mobs that travelled through a portal to be instantly despawned.

Rationale: Mob farms that [use portals](https://youtu.be/uh3X0Pmr9yc?t=904) to transport them to the overworld require 2 players, one in nether and one in the overworld. This is due to the hard despawn range limit of 128 for monsters.

gillesbraun committed:
Add setting to disable nerfing of portal based mob farms

gillesbraun committed:
Add setting to disable nerfing of portal based mob farms

gillesbraun committed:
Adjust commit message

NoahvdAa commented on patches/server/0879-Add-setting-to-disable-nerfing-of-portal-based-mob-f.patch:
```diff
@@ -0,0 +1,56 @@
+                 int j = i * i;
+ 
+-                if (d0 > (double) j) { // CraftBukkit - remove isTypeNotPersistent() check
++                // Paper - nerf portal mob farms
```
This comment should be on the line it changes (after CB)

gillesbraun replied:
Done

NoahvdAa reviewed (commented):
Welcome to Paper! :)

gillesbraun committed:
Add setting to disable nerfing of portal based mob farms
Marks entities that teleport through a nether portal with `fromNetherPortal`. Mobs with that flag that are prevented from despawning if the config value is set, effectively working around the hard entity despawn range limit. Defaults to the old behaviour of just deleting mobs that are beyond the hard range limit.

This allows mob farms that are set up in the nether to teleport mobs into the overworld, without them instantly despawning due to the hard despawn limit. Without this patch it requires another player to sit in the overworld to prevent the mobs from despawning.
